tests/conftest.py

```diff
@@ -234,43 +259,55 @@ def inner(
             wait = WebDriverWait(current_browser, 10)
-            if assert_success and retry:
+            if retry:
                 element = wait.until(EC.element_to_be_clickable((By.XPATH,
                                                                  "//input[contains(@id, 'passcode-input')]")))
             else:
-                element = wait.until(EC.visibility_of_element_located((By.XPATH, "//div[contains(text(), 'Bypass code') "
-                                                                                 "and "
-                                                                                 "contains( "
-                                                                       "@class, 'row') and contains(@class, "
-                                                                                 "'display-flex') ]"))
-                                 )
+                element = wait.until(EC.visibility_of_element_located((By.XPATH,
+                                                                       "//div[contains(text(), 'Bypass code') and "
+                                                                       "contains(@class, 'row') and contains(@class, "
+                                                                       "'display-flex')]")))
+
             current_browser.snap()
             element.click()
             current_browser.snap()
-            if assert_success and retry:
-                current_browser.execute_script("arguments[0].value = '';", element)
-                current_browser.snap()
+            if retry:
+                clear_passcode(current_browser, element)
             current_browser.send_inputs(passcode)
             current_browser.snap()
-            current_browser.snap()
             current_browser.wait_for_tag('button', 'Verify').click()
+            sleep(5)
+            current_browser.snap()
 
+            if test_env == 'eval' and is_this_your_device_screen:
+                if select_this_is_my_device:
+                    element = wait.until(EC.element_to_be_clickable((By.XPATH, "//button[@id='trust-browser-button']")))
+                else:
+                    element = wait.until(
+                        EC.element_to_be_clickable((By.XPATH, "//button[@id='dont-trust-browser-button' "
+                                                              "and text()='No, other people use this "
+                                                              "device']")))
+
+                element.click()
+            sleep(5)
             current_browser.snap()
 
         if assert_success:
             if test_env == 'prod':
                 current_browser.switch_to.default_content()
+            current_browser.snap()
             sp = sp_domain(match_service_provider) if match_service_provider else ''
             current_browser.wait_for_tag('h2', f'{sp} sign-in success!')
         elif assert_failure:
+            sleep(5)
```

Comment:
same ask as above (below?)

1. try using wait here if you haven't yet
2. add a comment indicating what's being waited for with the sleep

Reply:
removed